From c373e874a55ad5c82482cd0420d34e6c71efaaf1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sat, 10 Jun 2017 00:30:49 +0200
Subject: [PATCH] Allow configuration of full/current charge file names

---
 config.def.h | 8 +++++++-
 dwm-status.c | 4 ++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/config.def.h b/config.def.h
index 29db3f7..8667708 100644
--- a/config.def.h
+++ b/config.def.h
@@ -9,9 +9,15 @@ static const char *statsep = " | ";
 /* Format to use in the curtime function for the current time. */
 static const char *timefmt = "%d %b %Y %H:%M:%S";
 
-/* Path to power supply battery file in /sys. */
+/* Path to power supply battery directory in /sys. */
 static const char *sysbat = "/sys/class/power_supply/BAT0";
 
+/* Path to control file for current charge. */
+static const char *syscur = "charge_now"
+
+/* Path to control file for full charge. */
+static const char *sysfull = "charge_full_design";
+
 /* Sound card to use for alsa output. */
 static const unsigned int sndcrd = 0;
 
diff --git a/dwm-status.c b/dwm-status.c
index 968e385..e72e669 100644
--- a/dwm-status.c
+++ b/dwm-status.c
@@ -98,8 +98,8 @@ batcap(char *dest, size_t n)
 	size_t ret;
 	double res, curc, maxc;
 
-	curc = readnum((char*)sysbat, "charge_now");
-	maxc = readnum((char*)sysbat, "charge_full_design");
+	curc = readnum((char*)sysbat, (char*)syscur);
+	maxc = readnum((char*)sysbat, (char*)sysfull);
 
 	res = 100.0 * (curc / maxc);
 	if ((ret = snprintf(dest, n, "%.2f%%", res)) > n)
-- 
2.13.0

