diff -upr nldev-0.3_git20160208.orig/nldev.c nldev-0.3_git20160208/nldev.c
--- nldev-0.3_git20160208.orig/nldev.c	2016-02-10 22:55:42.918936670 +0100
+++ nldev-0.3_git20160208/nldev.c	2016-02-10 22:55:58.057382562 +0100
@@ -201,7 +201,7 @@ main(int argc, char *argv[])
 	showkernel = 1;
 	showudev = 1;
 	subsystem = NULL;
-	runpath = "/bin/mdev";
+	runpath = "/sbin/mdev";
 
 	ARGBEGIN {
 	case 'b':
