diff -upr nldev-0.3_git20160208.orig/run_nldev nldev-0.3_git20160208/run_nldev
--- nldev-0.3_git20160208.orig/run_nldev	2016-02-08 19:06:14.862154427 +0100
+++ nldev-0.3_git20160208/run_nldev	2016-02-08 19:06:34.730159003 +0100
@@ -6,5 +6,5 @@ then
 	ln -sf /proc/self/fd /dev/fd
 fi
 
-/bin/nldev -kd 2>&1 | logger -p daemon.notice -t nldev
+/usr/bin/nldev -kd 2>&1 | logger -p daemon.notice -t nldev
 
