diff -upr unbound-1.5.10.orig/doc/example.conf.in unbound-1.5.10/doc/example.conf.in
--- unbound-1.5.10.orig/doc/example.conf.in	2016-09-30 13:03:42.179328381 +0200
+++ unbound-1.5.10/doc/example.conf.in	2016-09-30 13:04:15.505829976 +0200
@@ -414,7 +414,7 @@ server:
 	# you start unbound (i.e. in the system boot scripts).  And enable:
 	# Please note usage of unbound-anchor root anchor is at your own risk
 	# and under the terms of our LICENSE (see that file in the source).
-	# auto-trust-anchor-file: "@UNBOUND_ROOTKEY_FILE@"
+	# auto-trust-anchor-file: ""
 
 	# File with DLV trusted keys. Same format as trust-anchor-file.
 	# There can be only one DLV configured, it is trusted from root down.
@@ -425,7 +425,7 @@ server:
 	# with several entries, one file per entry.
 	# Zone file format, with DS and DNSKEY entries.
 	# Note this gets out of date, use auto-trust-anchor-file please.
-	# trust-anchor-file: ""
+	trust-anchor-file: "@UNBOUND_ROOTKEY_FILE@"
 
 	# Trusted key for validation. DS or DNSKEY. specify the RR on a
 	# single line, surrounded by "". TTL is ignored. class is IN default.
Only in unbound-1.5.10/doc: example.conf.in.orig
