From af31655a5e8e2a936b19a791f4dd7ad6d66bf45b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Thu, 28 Jul 2022 20:15:19 +0200
Subject: [PATCH] tests: fix parse-cmd unit tests

---
 tests/parse-cmd.scm | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/tests/parse-cmd.scm b/tests/parse-cmd.scm
index e22bedd..5c1b7b7 100644
--- a/tests/parse-cmd.scm
+++ b/tests/parse-cmd.scm
@@ -1,9 +1,10 @@
-(import (edward ed))
+(import (edward ed cmd))
+(import (edward ed editor))
 
 (define (test-parse-cmd desc expected input)
   (test desc expected
         (let* ((cmd-input (string-append input "\n"))
-               (cmd-pair  (%test-parse parse-cmd cmd-input))
+               (cmd-pair  (%test-parse (parse-cmd) cmd-input))
                (cmd-addr  (car cmd-pair))
                (cmd-args  (cmd-args (cdr cmd-pair))))
           (append (list cmd-addr) cmd-args))))
@@ -42,7 +43,7 @@
       (list (addr->range (make-addr '(regex-backward . "foo") '(23 -42))))
       "foobar") "?foo? +23 -42 r foobar")
 
-  (test-parse-error "failed char pred" parse-cmd "rfoo"))
+  (test-parse-error "failed char pred" (parse-cmd) "rfoo"))
 
 (test-group "write command"
   (test-parse-cmd "no arguments"
@@ -57,7 +58,7 @@
               (make-addr '(current-line) '(10))))
       "foobar") ".,.+10w foobar")
 
-  (test-parse-error "failed char pred" parse-cmd "wfoo"))
+  (test-parse-error "failed char pred" (parse-cmd) "wfoo"))
 
 (test-group "shell command"
   (test-parse-cmd "no replacements"
@@ -117,6 +118,6 @@
           '())
      "42an\n.")
 
-  (test-parse-error "unknown command" parse-cmd "X")
-  (test-parse-error "failed char pred" parse-cmd "Qn")
-  (test-parse-error "failed char pred" parse-cmd "a n"))
+  (test-parse-error "unknown command" (parse-cmd) "X")
+  (test-parse-error "failed char pred" (parse-cmd) "Qn")
+  (test-parse-error "failed char pred" (parse-cmd) "a n"))
