From 654fed61cb459b1d8d0a6a418bd4a947461acd32 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Tue, 22 Nov 2016 17:06:37 +0100
Subject: [PATCH] Reset failure on XK_ESC and XK_BACKSPACE

---
 slock.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/slock.c b/slock.c
index d2f0886..2d03bbe 100644
--- a/slock.c
+++ b/slock.c
@@ -174,10 +174,12 @@ readpw(Display *dpy, struct xrandr *rr, struct lock **locks, int nscreens,
 			case XK_Escape:
 				explicit_bzero(&passwd, sizeof(passwd));
 				len = 0;
+				failure = 0;
 				break;
 			case XK_BackSpace:
 				if (len)
 					passwd[len--] = '\0';
+				failure = 0;
 				break;
 			default:
 				if (num && !iscntrl((int)buf[0]) &&
-- 
2.10.2

