From d7ce1891f5a36138e3cf3a2101efa442b20a5a0b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 17 Aug 2016 11:15:34 +0200
Subject: [PATCH 2/2] Reset failure on XK_ESC and XK_BACKSPACE

---
 slock.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/slock.c b/slock.c
index a00fbb9..6e820c7 100644
--- a/slock.c
+++ b/slock.c
@@ -169,10 +169,12 @@ readpw(Display *dpy, const char *pws)
 			case XK_Escape:
 				explicit_bzero(&passwd, sizeof(passwd));
 				len = 0;
+				failure = False;
 				break;
 			case XK_BackSpace:
 				if (len)
 					passwd[len--] = 0;
+				failure = False;
 				break;
 			default:
 				if (num && !iscntrl((int)buf[0]) && (len + num < sizeof(passwd))) {
-- 
2.9.3

