From d64efb7af316c0ed9f1c61ace480d7d40cfbc9b5 Mon Sep 17 00:00:00 2001
From: Duncaen <mail@duncano.de>
Date: Tue, 25 Sep 2018 16:37:07 +0200
Subject: [PATCH] mpick: add -v to usage

---
 mpick.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mpick.c b/mpick.c
index 57399db..b7fa595 100644
--- a/mpick.c
+++ b/mpick.c
@@ -1155,7 +1155,7 @@ main(int argc, char *argv[])
 		case 't': expr = chain(expr, EXPR_AND, parse_expr(optarg)); break;
 		case 'v': vflag = 1; break;
 		default:
-			fprintf(stderr, "Usage: %s [-T] [-t test] [msglist ...]\n", argv0);
+			fprintf(stderr, "Usage: %s [-Tv] [-t test] [msglist ...]\n", argv0);
 			exit(1);
 		}
 
