diff -upr dwm-6.0.orig/dwm.c dwm-6.0/dwm.c
--- dwm-6.0.orig/dwm.c	2011-12-19 16:02:46.000000000 +0100
+++ dwm-6.0/dwm.c	2015-10-11 03:09:27.322416202 +0200
@@ -736,9 +736,7 @@ drawbar(Monitor *m) {
 		           occ & 1 << i, urg & 1 << i, col);
 		dc.x += dc.w;
 	}
-	dc.w = blw = TEXTW(m->ltsymbol);
-	drawtext(m->ltsymbol, dc.norm, False);
-	dc.x += dc.w;
+	dc.x += blw;
 	x = dc.x;
 	if(m == selmon) { /* status is only drawn on selected monitor */
 		dc.w = TEXTW(stext);
